### ISSUE TITLE
feat: auto-layout

### ISSUE THREAD

**Issue by h0tw4t3r** [transmissions11/flux#39]
**Is your feature request related to a problem? Please describe.**
When you create a new node, it usually is created at the right of the last child node, which breaks the visual appearance of a tree.

**Describe the solution you'd like**
Add **automatic layout positioning** feature.

**Additional context**
- https://reactflow.dev/docs/examples/layout/auto-layout/

**Comment by transmissions11** [comment]
agreed auto layout would be a useful option (like auto zoom), but don't want to force it on everyone as some folks like dave like to layout their trees in funky ways.

**Comment by h0tw4t3r** [comment]
@transmissions11, absolutely! It makes sense to enable this on some key-hit, as it would be not effective to force it on every node addition.

**Comment by UnloadingGnat** [comment]
https://reactflow.dev/docs/examples/layout/dagre/
^ might be helpful
